Neos.Neos/Classes/Routing/FrontendNodeRoutePartHandler.php

```diff
@@ -574,19 +682,51 @@ protected function parseDimensionsAndNodePathFromRequestPathAllowingNonUniqueSeg
      * @param string $workspaceName Name of the workspace to use in the context
      * @param array $dimensionsAndDimensionValues An array of dimension names (index) and their values (array of strings). See also: ContextFactory
      * @return ContentContext
+     * @throws Exception\NoSiteException
      */
     protected function buildContextFromWorkspaceNameAndDimensions($workspaceName, array $dimensionsAndDimensionValues)
     {
         $contextProperties = [
             'workspaceName' => $workspaceName,
             'invisibleContentShown' => ($workspaceName !== 'live'),
             'inaccessibleContentShown' => ($workspaceName !== 'live'),
-            'dimensions' => $dimensionsAndDimensionValues
+            'dimensions' => $dimensionsAndDimensionValues,
+            'currentSite' => $this->getCurrentSite(),
         ];
 
         return $this->contextFactory->create($contextProperties);
     }
 
+    /**
+     * Determines the currently active site based on the "requestUriHost" parameter (that has to be set via HTTP middleware)
+     *
+     * @return Site
+     * @throws Exception\NoSiteException
+     */
+    protected function getCurrentSite(): Site
+    {
+        $requestUriHost = $this->parameters->getValue('requestUriHost');
+        if (!is_string($requestUriHost)) {
+            throw new Exception\NoSiteException('Failed to determine current site because the "requestUriHost" Routing parameter is not set', 1604860219);
+        }
+        if (!array_key_exists($requestUriHost, $this->siteByHostRuntimeCache)) {
```

Comment:
inversion of condition and early return would make this more readable and shave one level of indent off I think.

Reply:
good call, I extracted the actual site resolution to a dedicated method